frontend/src/modules/Dashboard/Components/Dashboard.tsx

```diff
@@ -28,17 +28,34 @@ type FilterOption =
   | 'matchable'
   | 'no-check-in-email'
   | 'no-no-match-email'
+const filterOptionDisplay = [
+  ['no-filter', 'All Classes'],
+  ['unmatchable', 'Unmatchable'],
+  ['newly-matchable', 'Newly Matchable'],
+  ['matchable', 'Matchable'],
+  ['no-check-in-email', 'No Check In Email'],
+  ['no-no-match-email', 'No No Match Email'],
+]
+const sortOrderDisplay = [
+  ['newest-requests-first', 'Newest Requests First'],
+  ['classes-a-z', 'Classes A-Z'],
+  ['classes-z-a', 'Classes Z-A'],
+]
 
 export const Dashboard = () => {
+  const history = useHistory()
   const { user } = useAuthValue()
   const { courses } = useCourseValue()
   const { students } = useStudentValue()
-
-  const [filteredOption, setFilteredOption] = useState<FilterOption>(
-    'no-filter'
+  const state = history.location.state as {
+    sortedOrder: SortOrder
```

Comment:
Great use of history here to store what you had for sort order and filter on the dashboard! It seems to work really well based off the videos you attached for testing. I also like what you did below by mapping elements of filterOptionDisplay instead of hard-coding the menu items – nice work there as well.